src/utils/attribute_authority.ts

```diff
@@ -37,8 +37,8 @@ export const getUserCompanies = (
 
     TE.chain((arr) => {
       return arr.length > 0
-        ? TE.left(ResponseErrorForbiddenNotAuthorized)
-        : TE.of(arr);
+        ? TE.of(arr)
+        : TE.left(ResponseErrorForbiddenNotAuthorized);
     }),
```

Comment:
What if we use a `fromPredicate` instead?
```suggestion
    TE.chainW(
      TE.fromPredicate(
        (_) => _.length > 0,
        () => ResponseErrorForbiddenNotAuthorized
      )
    ),
```

Reply:
Yes, I just fixed existing code, but you're right, it's a better solution to refactor since we are here.

Reply:
done in https://github.com/pagopa/hub-spid-login-ms/pull/54/commits/fd76da21c20a5323440aee87aa66c97e8e08323b 
I tried also for the ternary above this, but fromPredicate was not dividing the union of results...